app/scripts/modules/core/delivery/service/executions.transformer.service.js

```diff
@@ -255,12 +255,20 @@ module.exports = angular.module('spinnaker.core.delivery.executionTransformer.se
       }
     }
 
-    function addBuildInfo(execution) {
-      if (_.get(execution, 'trigger.parentExecution.trigger.buildInfo.number')) {
-        execution.buildInfo = execution.trigger.parentExecution.trigger.buildInfo;
-      }
+    function findNearestBuildInfo(execution) {
       if (_.get(execution, 'trigger.buildInfo.number')) {
-        execution.buildInfo = execution.trigger.buildInfo;
+        return execution.buildInfo = execution.trigger.buildInfo;
+      }
+      if (_.get(execution, 'trigger.parentExecution')) {
+        return findNearestBuildInfo(execution.trigger.parentExecution);
+      }
+      return false;
```

Comment:
The return values are a little ambiguous. Since you're already assigning the return value in `addBuildInfo`, it would read a little cleaner to just return the value or null, i.e.
```javascript
function findNearestBuildInfo(execution) {
  if (_.has(execution, 'trigger.buildInfo.number')) {
    return execution.trigger.buildInfo;
  }
  if (_.has(execution, 'trigger.parentExecution')) {
    return findNearestBuildInfo(execution.trigger.parentExecution);
  }
  return null;
}
```
Then, in `addBuildInfo`, you can omit the conditional block and just assign the value, i.e.
```javascript
function addBuildInfo(execution) {
  execution.buildInfo = findNearestBuildInfo(execution);
  ...
}
```